esphome/components/mopeka_pro_check/sensor.py

```diff
@@ -0,0 +1,133 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor, esp32_ble_tracker
+from esphome.const import (
+    CONF_DISTANCE,
+    CONF_MAC_ADDRESS,
+    CONF_ID,
+    ICON_THERMOMETER,
+    ICON_RULER,
+    UNIT_PERCENT,
+    CONF_LEVEL,
+    CONF_TEMPERATURE,
+    DEVICE_CLASS_TEMPERATURE,
+    UNIT_CELSIUS,
+    STATE_CLASS_MEASUREMENT,
+    CONF_BATTERY_LEVEL,
+    DEVICE_CLASS_BATTERY,
+)
+
+CONF_TANK_TYPE = "tank_type"
+CONF_CUSTOM_DISTANCE_FULL = "custom_distance_full"
+CONF_CUSTOM_DISTANCE_EMPTY = "custom_distance_empty"
+
+ICON_PROPANE_TANK = "mdi:propane-tank"
+
+TANK_TYPE_CUSTOM = "CUSTOM"
+
+UNIT_MILLIMETER = "mm"
+
+
+def small_distance(value):
+    """small_distance is stored in mm"""
+    meters = cv.distance(value)
+    return meters * 1000
+
+
+#
+# Map of standard tank types to their
+# empty and full distance values.
+# Format is - tank name: (empty distance in mm, full distance in mm)
+#
+CONF_SUPPORTED_TANKS_MAP = {
+    TANK_TYPE_CUSTOM: (0, 100),
+    "20LB_V": (38, 254),  # empty/full readings for 20lb US tank
+    "30LB_V": (38, 381),
+    "40LB_V": (38, 508),
+}
+
+CODEOWNERS = ["@spbrogan"]
+DEPENDENCIES = ["esp32_ble_tracker"]
+
+mopeka_pro_check_ns = cg.esphome_ns.namespace("mopeka_pro_check")
+MopekaProCheck = mopeka_pro_check_ns.class_(
+    "MopekaProCheck", esp32_ble_tracker.ESPBTDeviceListener, cg.Component
+)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(MopekaProCheck),
+            cv.Required(CONF_MAC_ADDRESS): cv.mac_address,
+            cv.Optional(CONF_CUSTOM_DISTANCE_FULL): small_distance,
+            cv.Optional(CONF_CUSTOM_DISTANCE_EMPTY): small_distance,
+            cv.Optional(CONF_TANK_TYPE, default="20LB_V"): cv.enum(
+                CONF_SUPPORTED_TANKS_MAP, upper=True
+            ),
```

Comment:
I actually noticed this stand out more when reading the docs PR, should there even be a default here? I think should be required for everyone.
```suggestion
            cv.Required(CONF_TANK_TYPE): cv.enum(
                CONF_SUPPORTED_TANKS_MAP, upper=True
            ),
```